utils-intervalrdd/src/main/scala/org/bdgenomics/utils/interval/array/IntervalArray.scala

```diff
@@ -65,7 +67,32 @@ object IntervalArray extends Serializable {
       }).sortByKey()
         .collect
 
-    new IntervalArray(sortedArray, accum.value, sorted = true)
+    buildFn(sortedArray, accum.value)
+  }
+
+  /**
+   * Sorts the RDD and collects it to build an IntervalArray, a sorted array that searches
+   * over ranges. This is used for a left side of the broadcast region join in ADAM.
+   *
+   * @param rdd RDD to build a IntervalArray from.
+   * @return The IntervalArray built from this RDD.
+   */
+  def apply[K <: Interval[K]: ClassTag, T: ClassTag](
+    rdd: RDD[(K, T)]): IntervalArray[K, T] = {
+
+    apply(rdd, apply(_, _, sorted = true))
+  }
+
+  def apply[K <: Interval[K]: ClassTag, T: ClassTag](
+    array: Array[(K, T)],
+    maxIntervalWidth: Long,
+    sorted: Boolean = true): IntervalArray[K, T] = {
+    val sortedArray = if (sorted) {
+      array
+    } else {
+      array.sortBy(_._1)
+    }
+    new ConcreteIntervalArray(sortedArray, maxIntervalWidth)
```

Comment:
`ConcreteIntervalArray` is not my favorite class name, but in these cases where there is only one implementation of a trait/interface, I'm never sure what to call it.  DefaultFoo, ConcreteFoo, FooImpl, etc.